### DISCUSSION TITLE
SQLite3 export

### DISCUSSION THREAD

**Discussion by mizaki** [GrandComicsDatabase/gcd-django#608] [General]
Hopefully this is a good place to ask?

I have been helping work on adding different information sources to [ComicTagger](https://github.com/comictagger/comictagger) and would like to add GCD to the list. First I presume there would be no issue with using the data (having read https://docs.comics.org/wiki/Data_Distribution and https://docs.comics.org/wiki/App_Guidelines) with CT?

To this end, I have looked at using the mysql dump that you offer. As expecting people to use MySQL for this seems unreasonable I used [this](https://github.com/dumblob/mysql2sqlite) script to convert it to SQLite. Obviously by the title, I wonder if it would be possible to get an SQLite alongside the MySQL dump? Asking users to use an awk script is also seems a bit much.

If you are amenable, I did discover [mysql-to-sqlite3](https://github.com/techouse/mysql-to-sqlite3
) in my search for a MySQL dump to SQLite tool. I don't know if it'd be any use or not.

If there is somewhere better to put this, please do let me know. Thanks.

**Comment by jochengcd** [comment]
The python-script would be preferred by us, do you have a chance to test it and if the resulting dump works for you ?

We likely can add it sqlite dump, if it makes it easier.

**Reply by mizaki** [reply]
I have tried the python mysql-to-sqlite3 (default options) with 2023-08-15.sql restored to MariaDB and it worked without issue. FYI it took about 20 minutes on an i5-4670K. The size does balloon from 2.4GB to 4.1GB. It appears to be working with CT too.

**Comment by mizaki** [comment]
Just checking back in. Is there anything more I can do to help with this?

**Comment by jochengcd** [comment]
I now added a sqlite-dump, using the awk-script in the end. This is to have the same data in the dump without handling the filtering of content in two places.

**Reply by mizaki** [reply]
Fantastic, thanks.
